File: maths/findNumbersWithEvenNumberOfDigits.java
package maths;
// https://leetcode.com/problems/find-numbers-with-even-number-of-digits/

class SolutionFindNumbersWithEvenNumbersOfDigit {
    public int findNumbers(int[] nums) {
        
        int count =0;
        for (int num : nums){
            if (num<0) num = num*-1; 
            if ( ((int) Math.log10(num)+1) %2==0) count++;
        }
        return count;  
    }
}
public class findNumbersWithEvenNumberOfDigits {
    public static void main(String[] args) {
        SolutionFindNumbersWithEvenNumbersOfDigit solution = new SolutionFindNumbersWithEvenNumbersOfDigit();
        int[] nums = {12,345,2,6,7896};
        System.out.println(solution.findNumbers(nums));

        int[] nums1 = {-23, 00, -343, -343483};
        System.out.println(solution.findNumbers(nums1));

    } 
}

File: maths/findNumbersWithEvenNumberOfDigits.py
# https://leetcode.com/problems/find-numbers-with-even-number-of-digits/

class Solution:
    def findNumbers(self, nums):
        
        #1 with string and len
        count=0
        for i in range(0,len(nums)):
            if (len(str(nums[i]))%2==0): count+=1
        return count
    
        '''
        #2 Using division
        count=0
        for i in nums:
            cur=0
            while(i>0):
                cur+=1
                i=i//10
            if cur%2==0: count+=1
        return count       
       
        #3 using log and floor 
        #floor(log10(num)) gives 1 value less than length of integer ie odd
        return sum((floor(log10(i)))%2 for i in nums)
        '''
